src/array.ts

```diff
@@ -283,6 +283,11 @@ export default class ArraySchema<
     }
     return base;
   }
+
+  describe(options?: ResolveOptions<TContext>) {
```

Comment:
can we combine the two here, and just keep the one override?

Reply:
Perhaps, but I struggled to understand how. I started off with something like
```ts
 describe(options?: ResolveOptions<TContext>) {
    const next = options ? this.resolve(options) : this.clone();
    let base = next.describe(options) as SchemaObjectDescription;
    ...
  }
``` 
But the problem is the `.resolve()` and `.clone()` both return a new schema. So that following `next.describe()` causes an infinite loop.

Reply:
I could do this
```ts
 describe(options?: ResolveOptions<TContext>) {
    const next = options ? this.resolve(options) : this.clone();
    let base = Schema.prototype.describe.call(next, options)  as SchemaObjectDescription
    ...
  }

```

Reply:
Another solution would be to remove this line: https://github.com/jquense/yup/blob/117843799e0fc3947f3583be7b60f309a25898d5/src/schema.ts#L908
And then add a `describe` method to every type that does the resolve/clone itself before calling `super.describe()`. This is explicit and seems OK too.

Which doe you prefer @jquense?
- Use `describe` and `_describe` as the code is now
- Use `Schema.prototype.describe.call`
- Define and explicit `describe` method on each type that is responsible for resolving itself

Reply:
what about:

```tsx
const next = this.resolve(options)
const base = this.withMutation(() => {
   return next.describe() // no options, shouldn't resolve
})

...
```

`withMutation` could be removed i am not sure why the root describe calls `clone()` maybe just to be safe with handing out mutable schema internals

Reply:
I believe calling `next.describe()` would cause an infinite loop. I think I have a good solution, and I have updated the PR @jquense.
The code now resolves these types twice in the `describe` method. For example
```js
const next = (options ? this.resolve(options) : this).clone();
const base = super.describe(options) as SchemaInnerTypeDescription;
```
Then we can use `next.whatever` instead of `this.whatever` to use the new resolved schema to add additional information to the description. I'm not really concerned about any performance overhead there. I don't think the `describe` method is generally being used in performance critical areas, and the `clone` and `resolve` should be fast anyway.